source/game/kart/KartObjectProxy.cc

```diff
@@ -317,6 +321,23 @@ bool KartObjectProxy::hasFloorCollision(const WheelPhysics *wheelPhysics) const
     return wheelPhysics->hitboxGroup()->collisionData().bFloor;
 }
 
+/// @addr{0x8058539C}
+std::pair<EGG::Vector3f, EGG::Vector3f> KartObjectProxy::cannonPosRot() {
+    const auto *cannon = System::CourseMap::Instance()->getCannonPoint(state()->cannonPointId());
+    EGG::Vector3f cannonPos = cannon->pos();
```

Comment:
Since `pos()` and `rot()` return `const &` we can skip a constructor call here by rewriting as 
```suggestion
    const EGG::Vector3f &cannonPos = cannon->pos();
    const EGG::Vector3f &cannonRot = cannon->rot();
```

Reply:
that is unfortunately not the case, bc at the end of the function we have
 ```cpp
    cannonPos += local60 * temp0;
    return std::pair(cannonPos, cannonRot);
```

Reply:
Could rewrite as
```
return std::pair(cannonPos + local60 * temp0, cannonRot);
```
@vabold opinions?

Reply:
"I would say that's better, yes" 
&nbsp;&nbsp;&nbsp;&nbsp;-- vabold, 2024